src/main/java/pl/patrycja/ox/ui/Sequence.java

```diff
@@ -0,0 +1,143 @@
+            int column = i % size;
+            if ((i + (unbrokenLine - 1) * size) > (size * size) - (size - column)) {
+                continue;
+            }
+            for (int j = 0; j < unbrokenLine; j++) {
+                stringBuilder.append((i + (j * size)) + 1).append(" ");
+                if (j != unbrokenLine - 1) {
+                    if (column < size - 1) {
+                        stringBuilder.append((i + (j * size)) + 1 + 1).append(" ");
+                    } else {
+                        stringBuilder.append(i + (j * size)).append(" ");
+                    }
+                }
+            }
+        }
+        writeToFile(stringBuilder, fileSequence);
+    }
+
+    private void generateDiagonalUp(int size, int unbrokenLine) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("X ");
+        for (int i = unbrokenLine - 1; i <= ((size - unbrokenLine) + 1) * size; i++) {
+            int column = i % size;
+            if (column < unbrokenLine - 1) {
+                continue;
+            }
+            for (int j = 0; j < unbrokenLine; j++) {
+                int k = i + (j * (size - 1));
+                stringBuilder.append(k + 1).append(" ");
+                if (j != unbrokenLine - 1) {
+                    if ((k % size) == size - 1) {
+                        stringBuilder.append(k).append(" ");
+                    } else {
+                        stringBuilder.append(k + 1 + 1).append(" ");
+                    }
+                }
+            }
+        }
+        writeToFile(stringBuilder, fileSequence);
+    }
+
+    private void generateDiagonalDown(int size, int unbrokenLine) {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("X ");
+        for (int i = 0; i <= (((size - unbrokenLine) + 1) * size) - unbrokenLine; i++) {
+            int column = i % size;
+            if (column > size - unbrokenLine) {
+                continue;
+            }
+            for (int j = 0; j < unbrokenLine; j++) {
+                int k = i + (j * (size + 1));
+                stringBuilder.append(k + 1).append(" ");
+                if (j != unbrokenLine - 1) {
+                    if ((k % size) == size - 1) {
+                        stringBuilder.append(k).append(" ");
+                    } else {
+                        stringBuilder.append(k + 1 + 1).append(" ");
+                    }
+                }
+            }
+        }
+        writeToFile(stringBuilder, fileSequence);
+    }
+
+    private void writeToFile(Object input, String pathToFile) {
+        File file = new File(pathToFile);
+        try {
+            BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file));
+            bufferedWriter.write(input.toString());
+            bufferedWriter.append("\n");
+            bufferedWriter.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+}
```

Comment:
Got it that logic is complicated for generating sequences but can we somehow mark those additional `1`s.
Also please extract repetable things to variables and reuse them.